cmd/codegen/generator/go/templates/src/_dagger.gen.go/client.go.tmpl

```diff
@@ -177,3 +177,84 @@ func (c errorWrappedClient) MakeRequest(ctx context.Context, req *graphql.Reques
 	}
 	return nil
 }
+{{/* If it's not in dev mode, we wrap the library client with custom options */}}
+{{ else }}
+
+func Connect(ctx context.Context, opts ...dagger.ClientOpt) (*Client, error) {
+	dag, err := dagger.Connect(ctx, opts...)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Client{
+		query:  dag.QueryBuilder(),
+		client: dag.GraphQLClient(),
+	}
+
+	if err := serveModuleDepencencies(ctx, c); err != nil {
+		return nil, err
+	}
+
+	return c, nil
+}
+{{ end }}
+
+// serveModuleDependencies services all dependencies of the module.
+// Local dependencies are served by the dagger.json.
+// Remote dependencies are generated by the client generator.
+func serveModuleDepencencies(ctx context.Context, client *Client) error {
+	{{ range $i, $dep := GitDependencies -}}
+	if err := client.ModuleSource(
```

Comment:
Is this bit still necessary? Isn't it included in `Dependencies` below?

Reply:
No, I'm still making the difference between the 2, local depends on dagger.json, but git doesn't, they are standalone dependencies

Reply:
Okay, but the git dependencies will be tried to be `Serve`d twice right if there *is* a `dagger.json`? Which is where the de-duplication logic comes in.

Reply:
Oh I see, yes, you're manually skipping over it. If you do don't skip over it... the de-duplication would apply right?

Reply:
That's why there's the check in the session, to verify that we never load 2 times the same module yes.